core/src/main/java/org/elasticsearch/index/query/IdsQueryBuilder.java

```diff
@@ -204,7 +204,7 @@ public String getWriteableName() {
     protected Query doToQuery(QueryShardContext context) throws IOException {
         Query query;
         if (this.ids.isEmpty()) {
-             query = Queries.newMatchNoDocsQuery();
+             query = Queries.newMatchNoDocsQuery("Missing ids in " + this.getName());
```

Comment:
nit: this will read "Missing ids in ids", might be a bit redundant. Not sure how to improve this, but maybe just quote it and add that it was a query?